### ISSUE TITLE
duplicate entries in tool search result set

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#10030]
This is today on .org, tried a couple of terms, every result hit seems duplicated 5 or more times

![Galaxy](https://user-images.githubusercontent.com/1814954/88412061-4526f600-cdd9-11ea-8fec-6e4a0985daab.png)

**Comment by martenson** [comment]
This substantially limits number of results and overall usefulness of the tool search.

**Comment by bgruening** [comment]
On EU we see 4 requests. I guess this correlates with the number of web workers?

**Comment by martenson** [comment]
@bgruening seems unclear since in the example above different hits have different number of duplicates

**Comment by bgruening** [comment]
Can you scroll down. In my case it is not sorted.

**Comment by martenson** [comment]
```
[
    "toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_sample_generator/sklearn_sample_generator/1.0.8.2",
    "toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_sample_generator/sklearn_sample_generator/1.0.8.2",
    "toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_sample_generator/sklearn_sample_generator/1.0.8.2",
    "toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_sample_generator/sklearn_sample_generator/1.0.8.2",
    "toolshed.g2.bx.psu.edu/repos/bgruening/sklearn_sample_generator/sklearn_sample_generator/1.0.8.2",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.26.0.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.26.0.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.26.0.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.26.0.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.26.0.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.0",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/devteam/sam_pileup/sam_pileup/1.1.3",
    "toolshed.g2.bx.psu.edu/repos/wolma/mimodd_main/mimodd_reheader/0.1.9",
    "toolshed.g2.bx.psu.edu/repos/wolma/mimodd_main/mimodd_reheader/0.1.9",
    "toolshed.g2.bx.psu.edu/repos/wolma/mimodd_main/mimodd_reheader/0.1.9",
    "toolshed.g2.bx.psu.edu/repos/wolma/mimodd_main/mimodd_reheader/0.1.9",
    "toolshed.g2.bx.psu.edu/repos/wolma/mimodd_main/mimodd_reheader/0.1.9",
    "toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.29.2"
]
```

**Comment by dannon** [comment]
Tracked down to recent changes to search index handling, fixed in #10061
